### PR TITLE
Support wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the package for testing against wagtail >= 4.1

Removes:

- Compatibility code and testing for wagtail < v4.1

Updates:

- docs

The test pass locally.

Todo:

- [ ] Test a wagtail 4.1 and 4.2 transfer scenario.

### CODE REVIEW

File: tests/settings.py [hunk 1]
```diff
@@ -154,3 +146,5 @@
 # 2.13.x (from 2.13.5 onward) and 2.14.x (from 2.14.2 onward) adopt the 2.15 behaviour, allowing us to
 # use the same test fixtures across all versions.
 WAGTAIL_COMMENTS_RELATION_NAME = 'wagtail_admin_comments'
```
Comment (victoriachan) [inline tests/settings.py:148]: Possiby can remove this too?

**Review by victoriachan — CHANGES_REQUESTED** [review]
Must be so satisfying! Just a few small points:

1) setup.py needs updating to remove python 3.6 and add python 3.11
2) settings.py: can remove WAGTAIL_COMMENTS_RELATION_NAME?

**Comment by nickmoreton** [comment]
> Must be so satisfying!

Yep but with so much to remove you think you have finished and mis a few things.

I have updated the comments thanks.  https://github.com/wagtail/wagtail-transfer/pull/144/commits/791fd2ec796a42078b52e56ad88f8975f590a30d & https://github.com/wagtail/wagtail-transfer/pull/144/commits/7493433a7174eb9fc1d1253137ed8b76fdb49a97

I did a little more adjustment to the Wagtail versions.

**Review by victoriachan — APPROVED** [review]
Looks good!

**Pull request merged**